dashboard/test/controllers/api/v1/teacher_feedbacks_controller_test.rb

```diff
@@ -0,0 +1,155 @@
+
+    teacher_sign_in_and_comment(@teacher, @student, @level, COMMENT1)
+    sign_out @teacher
+
+    teacher_sign_in_and_comment(teacher2, @student, @level, COMMENT2)
+    sign_out teacher2
+
+    sign_in @teacher
+    get "#{API}/show_feedback_from_teacher?student_id=#{@student.id}&level_id=#{@level.id}&teacher_id=#{@teacher.id}"
+    assert_equal COMMENT1, JSON.parse(@response.body)['comment']
+
+    get "#{API}/show_feedback_from_teacher?student_id=#{@student.id}&level_id=#{@level.id}&teacher_id=#{teacher2.id}"
+    assert_equal COMMENT2, JSON.parse(@response.body)['comment']
+  end
+
+  test 'retrieves comment on requested level when teacher has given student feedback on multiple levels' do
+    level2 = create :level
+    level3 = create :level
+
+    teacher_sign_in_and_comment(@teacher, @student, @level, COMMENT1)
+    teacher_sign_in_and_comment(@teacher, @student, level2, COMMENT2)
+    teacher_sign_in_and_comment(@teacher, @student, level3, COMMENT3)
+    get "#{API}/show_feedback_from_teacher?student_id=#{@student.id}&level_id=#{level2.id}&teacher_id=#{@teacher.id}"
+
+    assert_equal COMMENT2, JSON.parse(@response.body)['comment']
+  end
+
+  test 'retrieves the most recent comment from a teacher' do
+    teacher_sign_in_and_comment(@teacher, @student, @level, COMMENT1)
+    teacher_sign_in_and_comment(@teacher, @student, @level, COMMENT2)
+    teacher_sign_in_and_comment(@teacher, @student, @level, COMMENT3)
+    get "#{API}/show_feedback_from_teacher?student_id=#{@student.id}&level_id=#{@level.id}&teacher_id=#{@teacher.id}"
+
+    assert_equal COMMENT3, JSON.parse(@response.body)['comment']
+  end
+
+  test 'student can retrieve feedback for a level - two comments, one teacher' do
+    teacher_sign_in_and_comment(@teacher, @student, @level, COMMENT1)
+    teacher_sign_in_and_comment(@teacher, @student, @level, COMMENT2)
+    sign_out @teacher
+
+    sign_in @student
+    get "#{API}/show_feedback_for_level?student_id=#{@student.id}&level_id=#{@level.id}"
+
+    assert_equal 1, JSON.parse(@response.body)['feedbacks'].count
+    assert_equal COMMENT2, JSON.parse(@response.body)['feedbacks'][0]['comment']
+  end
+
+  test 'student can retrieve feedback for a level - three comments, two teachers' do
+    teacher2 = create :teacher
+    section2 = create :section, user: teacher2
+    section2.add_student(@student)
+
+    teacher_sign_in_and_comment(@teacher, @student, @level, COMMENT1)
+    sign_out @teacher
+    teacher_sign_in_and_comment(teacher2, @student, @level, COMMENT2)
+    sign_out teacher2
+    teacher_sign_in_and_comment(@teacher, @student, @level, COMMENT3)
+    sign_out @teacher
+
+    sign_in @student
+    get "#{API}/show_feedback_for_level?student_id=#{@student.id}&level_id=#{@level.id}"
+
+    assert_equal 2, JSON.parse(@response.body)['feedbacks'].count
+    assert_equal COMMENT2, JSON.parse(@response.body)['feedbacks'][1]['comment']
+    assert_equal COMMENT3, JSON.parse(@response.body)['feedbacks'][0]['comment']
```

Comment:
Suggestion: this can be DRY'd up and made a bit more readable by moving the parse into a private helper method, e.g.

```ruby
def parsed_response
  JSON.parse(@response.body)
end
```

Then it can be consumed like
```ruby
assert_equal 2, parsed_response['feedbacks'].count
```